traffic_ops/testing/api/v4/deliveryservices_test.go

```diff
@@ -1238,6 +1254,272 @@ func VerifyPaginationSupportDS(t *testing.T) {
+	qparams.Set("type", "10000")
+	resp, _, err := TOSession.GetDeliveryServicesV4(nil, qparams)
+	if err != nil {
+		t.Fatalf("Error!! Getting Deliveryservice by Invalid Type ID %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Fatalf("Error!! Invalid Type shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetDeliveryServiceByInvalidAccessibleTo(t *testing.T) {
+	qparams := url.Values{}
+	qparams.Set("accessibleTo", "10000")
+	resp, _, err := TOSession.GetDeliveryServicesV4(nil, qparams)
+	if err != nil {
+		t.Fatalf("Error!! Getting Deliveryservice by Invalid AccessibleTo %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Fatalf("Error!! Invalid AccessibleTo shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetDeliveryServiceByValidXmlId(t *testing.T) {
+	if len(testData.DeliveryServices) > 0 {
+		firstDS := testData.DeliveryServices[0]
+
+		resp, _, err := TOSession.GetDeliveryServiceByXMLIDNullableWithHdr(*firstDS.XMLID, nil)
+		if err != nil {
+			t.Fatalf("Error in Getting DeliveryServices by XML ID: %v - %v", err, resp)
+		}
+		if len(resp) == 0 {
+			t.Errorf("No delivery service available for the XML ID %v", *firstDS.XMLID)
+		} else {
+			if *resp[0].XMLID != *firstDS.XMLID {
+				t.Fatalf("Delivery Service Name expected: %s, actual: %s", *firstDS.XMLID, *resp[0].XMLID)
+			}
+		}
+	}
+}
+
+func GetDeliveryServiceByInvalidXmlId(t *testing.T) {
+	resp, _, err := TOSession.GetDeliveryServiceByXMLIDNullableWithHdr("test", nil)
+	if err != nil {
+		t.Fatalf("Error!! Getting Delivery service by Invalid ID %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Fatalf("Error!! Invalid Xml Id shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func SortTestDeliveryServices(t *testing.T) {
+	var header http.Header
+	var sortedList []string
+	resp, _, err := TOSession.GetDeliveryServicesV4(header, nil)
+	if err != nil {
+		t.Fatalf("Expected no error, but got %v", err.Error())
+	}
+	for i, _ := range resp {
+		sortedList = append(sortedList, *resp[i].XMLID)
+	}
+
+	res := sort.SliceIsSorted(sortedList, func(p, q int) bool {
+		return sortedList[p] < sortedList[q]
+	})
+	if res != true {
+		t.Errorf("list is not sorted by their XML Id: %v", sortedList)
+	}
+}
+
+func SortTestDeliveryServicesDesc(t *testing.T) {
+
+	var header http.Header
+	respAsc, _, err1 := TOSession.GetDeliveryServicesV4(header, nil)
+	params := url.Values{}
+	params.Set("sortOrder", "desc")
+	respDesc, _, err2 := TOSession.GetDeliveryServicesV4(header, params)
+
+	if err1 != nil {
+		t.Fatalf("Expected no error, but got error in DS Ascending %v", err1.Error())
+	}
+	if err2 != nil {
+		t.Fatalf("Expected no error, but got error in DS Descending %v", err2.Error())
+	}
+	// reverse the descending-sorted response and compare it to the ascending-sorted one
+	for start, end := 0, len(respDesc)-1; start < end; start, end = start+1, end-1 {
+		respDesc[start], respDesc[end] = respDesc[end], respDesc[start]
+	}
+	if !reflect.DeepEqual(respDesc[0].XMLID, respAsc[0].XMLID) {
```

Comment:
if `respDesc` *or* `respAsc` have zero length or are `nil`, or if `respDesc[0].XMLID` *or* `respAsc[0].XMLID` is/are `nil`, this will segfault

Reply:
Handled nil values.